src/index.ts

```diff
@@ -0,0 +1,52 @@
+import semanticIndexer, {
+    EmbeddingText,
+    SemanticIndexerOptions,
+    commonYargs,
+    CreateEmbeddingText,
+} from "@magda/semantic-indexer-sdk";
+import { existsSync, readFileSync } from "fs";
+import pdf2md from "@opendocsg/pdf2md";
+
+const args = commonYargs(6005, "http://localhost:6005");
+
+export const createEmbeddingText: CreateEmbeddingText = async ({
+    record,
+    format,
+    filePath,
+    url
+}) => {
+    if (format === "PDF" && filePath) {
+        if (!existsSync(filePath)) {
+            throw new Error(`File not found: ${filePath}`);
+        }
+
+        if (!filePath.endsWith(".pdf") && !filePath.endsWith(".PDF")) {
+            throw new Error(`Unexpected file format`);
+        }
+
+        const pdfBuffer = readFileSync(filePath);
+        const result = await pdf2md(pdfBuffer).catch((err) => {
```

Comment:
Since we use to markdown format here, we probably should use custom chucker --- could use 
MarkdownTextSplitter here:
https://github.com/langchain-ai/langchainjs/blob/ac42886f52a0bb851a281653a77baee019725ea9/libs/langchain-textsplitters/src/text_splitter.ts#L777C14-L777C34